### ISSUE TITLE
Compare "fake"-store before and after running saga

### ISSUE THREAD

**Issue by JulianMeinhardt** [jfairbank/redux-saga-test-plan#120]
Hi,

I'm currently trying some things out to find a good test framework for the sagas in our current project.
Currently i'm only testing for the effects of a saga. But I also want to do check the changes in our global redux store like this:
```javascript
const HAVE_BIRTHDAY = 'HAVE_BIRTHDAY';

const initialState = {
  dog: {
    name: 'Tucker',
    age: 11,
  },
};

function reducer(state = initialState, action) {
  if (action.type === HAVE_BIRTHDAY) {
    return {
      ...state,
      dog: {
        ...state.dog,
        age: state.dog.age + 1,
      },
    };
  }

  return state;
}

const store = createStore(reducer, initState);

function* saga(api) {
  // It could do some more stuff (e.g doing API calls)
  // but the important thing is that it dispatches an action against
  // the store which states changes after that
  yield put({type: 'HAVE_BIRTHDAY'});
}

it('handles generators', () => {

  return expectSaga(saga, api)
    // add store for the saga
    .withStore(store)

    .put({ type: HAVE_BIRTHDAY })

    .run()
    // then i would love to check for the current state
    .getStore().to.have.deep.property('dog.age',12);
});
```

There is already a with Reducer method on redux-saga-test-plan. But no way to get the current state after the sagas or is it?

**Comment by jfairbank** [comment]
Hi, @JulianMeinhardt!

Unfortunately, there isn't a way to check the store state. However, it would be trivial to implement. I'm finishing up #114, so I can easily expose the final store state in the next release to allow you to make assertions against it. Something like this:

```js
it('handles generators', () => {
  return expectSaga(saga, api)
    .withReducer(reducer)
    .put({ type: HAVE_BIRTHDAY })

    .run() // <-- .run() returns a Promise

    .then(({ storeState }) => {
      expect(storeState.dog.age).toBe(12);
    });
});

**Comment by jfairbank** [comment]
This is available in v3.1.0 now via the exposed store state in the `Promise` and an assertion method called `hasFinalState`.

```js
it('exposes state', () => {
  return expectSaga(saga, api)
    .withReducer(reducer)
    .put({ type: HAVE_BIRTHDAY })

    .run()

    .then(({ storeState }) => {
      expect(storeState.dog.age).toBe(12);
    });
});

it('can assert state', () => {
  return expectSaga(saga, api)
    .withReducer(reducer)
    .put({ type: HAVE_BIRTHDAY })

    .hasFinalState({
      dog: {
        name: 'Tucker',
        age: 12,
      },
    })

    .run();
});
```
